migration.rst

```diff
@@ -272,24 +291,110 @@ after Symfony finished handling the request you reduced the chances
+also allow you to incrementally improve the legacy code making it easier
+to transition it over to the new Symfony application.
+
 The major downside is, that both systems are not well integrated
 into each other leading to some redundancies and possibly duplicated code.
 For example since the Symfony application is already done handling the
 request you can not take advantage of kernel events, utilize Symfony's
 routing for determining which legacy script to call.
 
 
-Legacy Router Provider
-......................
+Legacy Route Loader
+...................
+
+The major difference to the LegacyBridge-approach from before is, that we
+move the logic inside the Symfony application. Removing some of the
+redundancies and allowing us to also interact with parts of the legacy
+application from inside Symfony, instead of just the other way around.
+
+.. tip::
 
-.. TODO: Use LegacyRouteProvider to fall back to legacy application
+    The following route loader is just a generic example that you might
+    have to tweak for your legacy application. You can familiarize
+    yourself with the concepts of this loader by reading up on it in
+    the :ref:`routing/custom_route_loader` section.
 
+The legacy route loader has a similar functionality as the previous
+LegacyBridge, but it is a service that is registered inside Symfony's
+routing component::
 
-Wrapping Legacy Application in a Controller
-...........................................
+    public function load($resource, $type = null)
+    {
+        $collection = new RouteCollection();
+        $finder = new Finder();
+        $finder->files()->name('*.php');
+
+        /** @var SplFileInfo $legacyScriptFile */
+        foreach ($finder->in($this->webDir) as $legacyScriptFile) {
+            // This assumes all legacy files use ".php" as extension
+            $filename = basename($legacyScriptFile->getRelativePathname(), '.php');
+            $routeName = sprintf('app.legacy.%s', str_replace('/', '__', $filename));
+
+            $collection->add($routeName, new Route($legacyScriptFile->getRelativePathname(), [
+                '_controller' => 'App\Controller\LegacyController::loadLegacyScript',
+                'requestPath' => '/' . $legacyScriptFile->getRelativePathname(),
+                'legacyScript' => $legacyScriptFile->getPathname(),
+            ]));
+        }
 
-.. TODO: Stream response from the legacy application in a controller action
+        return $collection;
+    }
+
+This only shows the ``load``-method of a custom Route Loader. You will also
+have to register the loader in your application's ``routing.yaml`` as
+described in the documentation for :ref:`routing/custom_route_loader`<Custom Route Loaders>.
+Depending on your configuration you might also have to tag the service with
+``routing.loader``. Afterwards you should be able to see all the legacy routes
+in your route configuration, e.g. when you call the ``debug:router``-command::
```

Comment:
Pleas use „.. code-block:: terminal“ instead of „::“ which is only for php syntax highlighting